database/migrations/sqls/20240617151644-pgwar-community-entity-labels-up.sql

```diff
@@ -0,0 +1,150 @@
+        entity_preview.fk_class IS DISTINCT FROM EXCLUDED.fk_class OR
+        entity_preview.entity_type IS DISTINCT FROM EXCLUDED.entity_type OR
+        entity_preview.class_label IS DISTINCT FROM EXCLUDED.class_label OR
+        entity_preview.entity_label IS DISTINCT FROM EXCLUDED.entity_label OR
+        entity_preview.full_text IS DISTINCT FROM EXCLUDED.full_text OR
+        entity_preview.ts_vector IS DISTINCT FROM EXCLUDED.ts_vector OR
+        entity_preview.type_label IS DISTINCT FROM EXCLUDED.type_label OR
+        entity_preview.fk_type IS DISTINCT FROM EXCLUDED.fk_type OR
+        entity_preview.time_span IS DISTINCT FROM EXCLUDED.time_span OR
+        entity_preview.first_second IS DISTINCT FROM EXCLUDED.first_second OR
+        entity_preview.last_second IS DISTINCT FROM EXCLUDED.last_second;
+END;
+$$
+LANGUAGE plpgsql;
+
+-- Update community entity labels on change on projects entity labels
+CREATE OR REPLACE FUNCTION pgwar.update_community_entity_label_on_project_entity_label_change()
+    RETURNS TRIGGER AS $$
+DECLARE
+    label text;
+BEGIN
+    IF NEW.fk_project != 0 OR OLD.fk_project !=0 THEN
+
+        IF tg_op = 'UPDATE' AND OLD.entity_label != NEW.entity_label THEN
+            SELECT pgwar.get_most_frequent_entity_label(NEW.pk_entity) INTO label;
+            PERFORM
+                pgwar.upsert_community_entity_preview((
+                   NEW.pk_entity,
+                   0,
+                   NEW.fk_class,
+                   NEW.entity_type,
+                   NEW.class_label,
+                   label,
+                   NEW.full_text,
+                   NEW.ts_vector,
+                   NEW.type_label,
+                   NEW.fk_type,
+                   NEW.time_span,
+                   NEW.first_second,
+                   NEW.last_second,
+                   NEW.tmsp_last_modification)::pgwar.entity_preview
+                );
+        ELSEIF tg_op = 'INSERT' THEN
+            SELECT pgwar.get_most_frequent_entity_label(NEW.pk_entity) INTO label;
+            PERFORM
+                pgwar.upsert_community_entity_preview((
+                   NEW.pk_entity,
+                   0,
+                   NEW.fk_class,
+                   NEW.entity_type,
+                   NEW.class_label,
+                   label,
+                   NEW.full_text,
+                   NEW.ts_vector,
+                   NEW.type_label,
+                   NEW.fk_type,
+                   NEW.time_span,
+                   NEW.first_second,
+                   NEW.last_second,
+                   NEW.tmsp_last_modification)::pgwar.entity_preview
+                );
+        ELSE
+            SELECT pgwar.get_most_frequent_entity_label(OLD.pk_entity) INTO label;
+            UPDATE pgwar.entity_preview
+            SET entity_label = label
+            WHERE fk_project = 0 AND pk_entity = OLD.pk_entity;
+        END IF;
+    END IF;
+
+    RETURN NULL;
+END;
+$$ LANGUAGE plpgsql;
+
```

Comment:
I suggest to simplify the function `pgwar.update_community_entity_label_on_project_entity_label_change()` so:

```suggestion
-- Update community entity labels on change on projects entity labels
CREATE OR REPLACE FUNCTION pgwar.update_community_entity_label_on_project_entity_label_change()
    RETURNS TRIGGER AS $$
DECLARE
    entity_id int;
    label text;
BEGIN
    entity_id := COALESCE(NEW.pk_entity, OLD.pk_entity);

    PERFORM pgwar.get_and_update_community_entity_label(entity_id);
    
    RETURN NULL;
END;
$$ LANGUAGE plpgsql;

```

Reply:
resolved in #363